### PR TITLE
Make the Docker image cross-compile without QEMU emulation

### PR DESCRIPTION
**Pull request by sandhose**
This does three things:

- Bump all dependencies & disable the pyo3 feature by default (because I had issues with old versions of maturin & openssl)
- Make the Docker image cross-compile without QEMU emulation
- Add a build cache on the Docker image

### CODE REVIEW

File: synapse_auto_compressor/pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+[build-system]
```
Comment (clokep) [inline synapse_auto_compressor/pyproject.toml:1]: I guess since this is essentially a separate package we need a different pyproject for it?

**Review by clokep — COMMENTED** [review]
I'm not really sure how to test this -- should we kick off a manual build to see it work? (I'm unsure if that'll push the image / cause it to get used automatically.)

Overall it looks plausible though.

**Comment by sandhose** [comment]
> I'm not really sure how to test this -- should we kick off a manual build to see it work? (I'm unsure if that'll push the image / cause it to get used automatically.)

You can try building the image locally for a foreign platform, e.g. `docker build --platform=linux/arm64 .`, and it should work if you have a recent enough docker?

File: synapse_auto_compressor/pyproject.toml (hunk 1 above)
Comment (sandhose) [reply synapse_auto_compressor/pyproject.toml:1]: I don't remember why I did this exactly, but I think that when I tried to upgrade maturin/pyo3, it was starting to become an issue that they were not properly two different projects?

**Review by clokep — APPROVED** [review]
Seemed to work OK.

**Pull request merged**